staging/src/k8s.io/client-go/util/workqueue/default_rate_limiters_test.go

```diff
@@ -191,7 +191,7 @@ func TestWithMaxWaitRateLimiter(t *testing.T) {
 		}
 	}
 
-	for i := 100; i < 5100; i++ {
+	for i := 100; i <= 5100; i++ {
```

Comment:
Hi, @Karthik-K-N, thanks for looking into the flake! I don't think this is enough to fix the flake though. Because the `DefaultControllerRateLimiter` uses a real-time clock underneath, if this test gets suspended by the golang scheduler when executing line 195, we can still get a `<500s` return value at line 201. 

I think instead of using `DefaultControllerRateLimiter`, we should make a fake limiter, which returns steadily increasing delay every time `When()` is called, and we verify that `MaxWaitRateLimiter` respects the 500s maximum even if the fake limiter returns >500s delay.

Reply:
Thanks @caesarxuchao for your inputs, I have added a new commit with a simple fake rate limiter which will either return base delay duration or max delay based on the limit set, This will help to test the `MaxWaitRateLimiter`. Could you please take a look